### ISSUE TITLE
Failed nodes are no longer preserved / don't show up in recent failures

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#13586]
My nodes are continuously failing to start:
```
(autoscaler +38s) Adding 1 nodes of type gpu.medium.
(autoscaler +49s) Removing 1 nodes of type gpu.medium (launch failed).
(autoscaler +54s) Adding 1 nodes of type gpu.medium.
(autoscaler +1m6s) Removing 1 nodes of type gpu.medium (launch failed).
(autoscaler +1m11s) Adding 1 nodes of type gpu.medium.
(autoscaler +1m22s) Removing 1 nodes of type gpu.medium (launch failed).
(autoscaler +1m27s) Adding 1 nodes of type gpu.medium.
(autoscaler +1m39s) Removing 1 nodes of type gpu.medium (launch failed).
(autoscaler +1m44s) Adding 1 nodes of type gpu.medium.
(autoscaler +1m55s) Removing 1 nodes of type gpu.medium (launch failed).
(autoscaler +2m0s) Adding 1 nodes of type gpu.medium.
(autoscaler +2m11s) Adding 1 nodes of type gpu.medium.
(autoscaler +2m23s) Removing 1 nodes of type gpu.medium (launch failed).
```

However, ray status only shows
```
======== Autoscaler status: 2021-01-20 09:46:27.976555 ========
Node status
---------------------------------------------------------------
Healthy:
 1 anyscale.cpu.medium
Pending:
 172.31.31.155: anyscale.gpu.medium, setting up
Recent failures:
 (no failures)

Resources
---------------------------------------------------------------

Usage:
 0.0/16.0 CPU
 0.00/36.670 GiB memory
 0.00/12.646 GiB object_store_memory

Demands:
 {'GPU': 1.0, 'CPU': 1.0}: 1+ pending tasks/actors
```

This seems to be since the failed nodes aren't preserved in the status. cc @wuisawesome @AmeerHajAli

**Comment by AmeerHajAli** [comment]
I pointed this out already in @wuisawesome 's logging PR that failed tests won't be tracked.

**Comment by wuisawesome** [comment]
Yes, we left it as a TODO in the code.

**Comment by ericl** [comment]
It used to be tracked, since we didn't terminate failed nodes. Isn't this a usability regression since users used to be able to debug failed nodes by ssh'ing in manually? Now it seems no longer possible.

**Comment by wuisawesome** [comment]
@DmitriGekhtman once we track the node updater's output, this should become trivial (the set of failed nodes = (set of all nodes) - (set of healthy nodes) )

**Comment by ericl** [comment]
fyi @clarkzinzow this might be a good starter issue

**Comment by AmeerHajAli** [comment]
Closed in #14608. @wuisawesome, please reopen if I am wrong.
